Cleanup to_zarr docstring

djhoese opened the pull request:
Talking with @jrbourbeau during the summit and I was confused that `to_zarr` does have a `return_stored` kwarg because I wasn't seeing it in the docstring. It is actually in the docstring, but was formatted poorly so it was combined with the compute kwarg. This PR reformats the docstring so things are rendered properly in the HTML.

Question: How does the dask project feel about API references (and intersphinx) in the docstrings? I also added a See Also for the store function, but it doesn't actually link to the function's documentation. This could be possible by doing:

```
:func:`da.store <dask.array.core.store>`
```

I understand this is a little ugly and may not be what you're going for in the docs, but it is helpful to the user so they don't have to go search for the function in the API docs.

- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask` / `isort dask`

jrbourbeau commented on dask/array/core.py:
```diff
@@ -2918,8 +2918,12 @@ def to_zarr(
-    compute, return_stored: see ``store()``
-    kwargs: passed to the ``zarr.create()`` function, e.g., compression options
+    compute: bool
+        See ``store()``.
```
I realize this predates this PR, but maybe we something like "See ``store()`` for more details." would be slightly more clear. Additionally, feel free to link directly to the `store` API docs here

jrbourbeau commented on dask/array/core.py:
```diff
@@ -2918,8 +2918,12 @@ def to_zarr(
+    compute: bool
+        See ``store()``.
+    return_stored: bool
+        See ``store()``.
```
Similar comment here

jrbourbeau reviewed (commented):
Thanks @djhoese! Spreading these out to separate lines seems more readable to me. 

> How does the dask project feel about API references (and intersphinx) in the docstrings?

This is definitely something we utilize. You can do `git grep ":func:"` in the `dask` repo to see all the places we link out to other API docs directly.

djhoese commented:
Ok, had my `main` branch renamed but forgot to pull the most recent changes from upstream. Updated, forced pushed now and we'll see how things go. As for the actual changes...

The references all work now, but for some reason when I ran this locally the anchor to the dask.array.store function got rewritten as `#id41` or something. I'll have to test again now that I've rebased.

djhoese commented:
Oh I also added zarr to the intersphinx in conf.py.

djhoese commented:
> I'll have to test again now that I've rebased.

False alarm. This looks good now.

jrbourbeau commented on dask/array/core.py:
```diff
@@ -3315,7 +3319,8 @@ def to_zarr(
     See Also
     --------
-    dask.array.Array.compute_chunk_sizes
+    :func:`dask.array.store`
+    :meth:`dask.array.Array.compute_chunk_sizes`
```
Hmm I wonder if `:func:` and `:meth:` are needed in "See Also" sections? Looking at the [current `to_zarr` docstring](https://docs.dask.org/en/latest/array-api.html#dask.array.to_zarr) it appears `dask.array.Array.compute_chunk_sizes` is already a hyperlink to the corresponding docstring

TomAugspurger replied:
They are not required (numpydoc handles it automatically).

jrbourbeau replied:
Nice, good to know! @djhoese would you mind removing the extra `:func:` and `:meth:` parts? After that, this should be good to merge

djhoese replied:
Thanks. That was left over when I was going to try to see if it made sense to shorten the link text, but then changed my mind. This should not be fixed.

jrbourbeau reviewed (commented):
Thanks for the updates @djhoese! I left one small question about the "See Also" section, but otherwise this looks great

jrbourbeau reviewed (approved):
Thanks @djhoese! This is in
